modules/dnn/src/layers/einsum_layer.cpp

```diff
@@ -0,0 +1,1200 @@
+        // this will hold input dims after reordering so that all inputs have
+        // same axes order
+        MatShape homogenizedInputDims_(numLetterIndices, 1);
+
+        int dimIndexInIreprocessedInput = 0;
+        int dimIndexInOriginalInput = 0;
+
+        for (const auto& subscriptIndex : currSubscriptIndices)
+        {
+            if(subscriptIndicesToInputIndex[subscriptIndex] == -1)
+            {
+                subscriptIndicesToInputIndex[subscriptIndex] = dimIndexInIreprocessedInput++;
+                homogenizedInputDims_[subscriptIndex] = input_dims[dimIndexInOriginalInput];
+            } else {
+                // Call diagonal
+                preprocessed = Diagonal(
+                    !preprocessed.empty() ? preprocessed : inputs[inputIter],
+                    subscriptIndicesToInputIndex[subscriptIndex],
+                    dimIndexInIreprocessedInput);
+            }
+            ++dimIndexInOriginalInput;
+        }
+
+        std::vector<size_t> permutation;
+        for(auto& d : subscriptIndicesToInputIndex)
+        {
+            if (d != -1)
+                permutation.emplace_back(d);
+        }
+
+        if (IsTransposeRequired(
+            !preprocessed.empty() ? preprocessed.size.dims() : inputs[inputIter].size.dims(),
+            permutation))
+        {
+            // call transpose
+            preprocessed = Transpose(
+                !preprocessed.empty() ? preprocessed : inputs[inputIter],
+                !preprocessed.empty() ? shape(preprocessed) : shape(inputs[inputIter]),
+                permutation);
+        }
+
+        if (!preprocessed.empty())
+        {
+            // check if this is correct
+            preprocessed = preprocessed.reshape(1, homogenizedInputDims_.size(), homogenizedInputDims_.data());
+        }
+
+        // fails here! check the problem
+        preProcessedInputs.emplace_back(preprocessed);
+        homogenizedInputDims.emplace_back(homogenizedInputDims_);
+
+        ++inputIter;
+    }
+}
+
+void LayerEinsumImpl::parseEquation(const String& equation)
+{
+    // copy copy of an eqution, will be changed
+    String eq = equation;
+
+    // remove white spaces in the copy
+    eq.erase(std::remove_if(eq.begin(), eq.end(), ::isspace), eq.end());
+
+    // check if '->' - the output subscript label is present in the equation;
+    std::size_t arrow_idx = eq.find("->");
+    if (arrow_idx != std::string::npos)
+    {
+        // split left and righ hand sides of the equation
+        lhs_eq = eq.substr(0, arrow_idx);
+        rhs_eq = eq.substr(arrow_idx + 2);
+        explicitEquation = true;
+    } else {
+        lhs_eq = eq;
+    }
+
+    // split lhs_eq by ',' - comma and put all created token - splits
+    // into lhs_eq_tokens vector
+    String token, comma = ",";
+    size_t idx = 0;
+    while(idx != String::npos){
+        idx = lhs_eq.find(comma);
+        token = lhs_eq.substr(0, idx);
+        lhs_eq.erase(0, idx + comma.length());
```

Comment:
Do you really need to `erase`? You can jsy increment the first parameter in the line above.